deft.el

```diff
@@ -981,7 +986,29 @@ See `deft-find-all-files'."
                  (not (backup-file-name-p file))
                  (member (file-name-extension file) deft-extensions))
             (setq result (cons file result)))))
-        result)))
+        (deft-apply-generation-rules result))))
+
+(defun deft-apply-generation-rules (lst)
+  (let ((result nil))
+    (when deft-generation-rules
+    (dolist (file lst)
+      (when (not (deft-exported-from-previous? file deft-generation-rules result))
```

Comment:
Indeed, I tested this with `(setq deft-generation-rules '(("txt" . "tex")))` and since `tex` is before `txt` in alphabetical order, the `tex` file was not excluded. Unless you disagree, I will apply your patch and fix this issue at the same time. (I have already done it locally, but have not pushed it yet, so no need to update your PR.)